pecan/routing.py

```diff
@@ -170,18 +173,16 @@ def lookup_controller(obj, remainder, request=None):
 def handle_lookup_traversal(obj, args):
     try:
         result = obj(*args)
+    except TypeError as te:
+        logger.debug('Got exception calling lookup(): %s (%s)',
+                     te, te.args)
+    else:
```

Comment:
You want to log if there's a problem (`TypeError`) when calling the `obj`. If any of the other line raises a `TypeError`, it'd be a bug in Pecan. You don't want to catch that.

Yes it's unlikely the few lines will raise in `TypeError` but there's no good reason to keep them in the except block.